tests/lib/lte_lc_api/src/lte_lc_api_test.c

```diff
@@ -0,0 +1,1261 @@
+		strcpy(pattern_buf[2], nrf_modem_at_scanf_return.pattern_buf[2]);
+		strcpy(pattern_buf[3], nrf_modem_at_scanf_return.pattern_buf[3]);
+
+		result = 4;
+		break;
+	default:
+		/* function called more often than expected */
+		TEST_FAIL_MESSAGE("nrf_modem_at_scanf called more times than expected");
+		return -EPERM;
+	}
+
+	/* advance queue */
+	for (size_t i = 1; i < ARRAY_SIZE(nrf_modem_at_scanf_variant); ++i) {
+		nrf_modem_at_scanf_variant[i-1] = nrf_modem_at_scanf_variant[i];
+	}
+	return result;
+}
+
+void wrap_lc_init(void)
+{
+	nrf_modem_at_scanf_return.ltem_mode = 1;
+	nrf_modem_at_scanf_return.gps_mode = 1;
+	nrf_modem_at_scanf_return.nbiot_mode = 1;
+	nrf_modem_at_scanf_return.mode_preference = 0;
+	nrf_modem_at_scanf_variant[0] = XSYSTEMMODE_lte_lc_system_mode_get;
+	nrf_modem_at_scanf_variant[1] = INVALID;
+
+	__wrap_nrf_modem_at_printf_ExpectAndReturn("AT%%XSYSTEMMODE=%s,%c", EXIT_SUCCESS);
+	__wrap_nrf_modem_at_printf_ExpectAndReturn("AT+CEREG=5", EXIT_SUCCESS);
+	__wrap_nrf_modem_at_printf_ExpectAndReturn("AT+CSCON=1", EXIT_SUCCESS);
+	int ret = lte_lc_init();
+
+	TEST_ASSERT_EQUAL(EXIT_SUCCESS, ret);
+}
+
+void setUp(void)
+{
+	mock_functions_to_mock_Init();
+	wrap_lc_init();
+}
+
+void tearDown(void)
+{
+	int ret;
+
+	__wrap_nrf_modem_at_printf_ExpectAndReturn("AT+CFUN=%d", EXIT_SUCCESS);
+
+	TEST_ASSERT_EQUAL_MESSAGE(INVALID, nrf_modem_at_scanf_variant[0],
+		"nrf_modem_at_scanf called fewer times than expected");
+	ret = lte_lc_deinit();
+	TEST_ASSERT_EQUAL(EXIT_SUCCESS, ret);
+	mock_functions_to_mock_Verify();
+}
+
+/* This is needed because AT Monitor library is initialized in SYS_INIT. */
+static int sys_init_helper(const struct device *unused)
+{
+	__wrap_nrf_modem_at_notif_handler_set_ExpectAnyArgsAndReturn(0);
+
+	return 0;
+}
+
+
+void test_lte_lc_register_handler_null(void)
+{
+	lte_lc_register_handler(NULL);
+}
+
+void test_lte_lc_deregister_handler_null(void)
+{
+	int ret;
+
+	ret = lte_lc_deregister_handler(NULL);
+	TEST_ASSERT_EQUAL(-EINVAL, ret);
+}
+
+void test_lte_lc_deregister_handler_twice(void)
+{
+	int ret;
+
+	lte_lc_register_handler((void *) 1);
+	lte_lc_register_handler((void *) 1);
+	ret = lte_lc_deregister_handler((void *) 1);
+	TEST_ASSERT_EQUAL(EXIT_SUCCESS, ret);
+}
+
+void test_lte_lc_deregister_handler_not_initialized(void)
+{
+	int ret;
+
+	__wrap_nrf_modem_at_printf_ExpectAndReturn("AT+CFUN=%d", EXIT_SUCCESS);
+	ret = lte_lc_deinit();
+	TEST_ASSERT_EQUAL(EXIT_SUCCESS, ret);
+
+	ret = lte_lc_deregister_handler(NULL);
+	TEST_ASSERT_EQUAL(-EINVAL, ret);
+
+	/* this is just to make the tearDown work again */
+	wrap_lc_init();
+}
+
+void test_lte_lc_deinit_twice(void)
+{
+	int ret;
+
+	__wrap_nrf_modem_at_printf_ExpectAndReturn("AT+CFUN=%d", EXIT_SUCCESS);
+	ret = lte_lc_deinit();
+	TEST_ASSERT_EQUAL(EXIT_SUCCESS, ret);
+	ret = lte_lc_deinit();
+	TEST_ASSERT_EQUAL(EXIT_SUCCESS, ret);
+
+	/* this is just to make the tearDown work again */
+	wrap_lc_init();
+}
+
+void test_lte_lc_connect_home(void)
```

Comment:
This and the following test case (`test_lte_lc_init_and_connect_home()`) don't actually connect to the network, because the response indicates that modem is already registered. This may be intentional, but maybe it could be visible in the test case name or in a comment.

Reply:
done